R/commonly_used_db_data.R

```diff
@@ -0,0 +1,277 @@
+                              AND scenario_type IN %s
+                              AND gavi_support_level IN %s
+                              AND vaccine NOT IN %s",
+                                                 sql_in(scenario_type),
+                                                 sql_in(gavi_support_levels), 
+                                                 sql_in(vaccine_to_ignore)),
+                                         "AND scenario_description NOT LIKE '%LiST%'"),
+                              touchstone_cov)
+  if(nrow(cov_sets) == 0L){
+    # this is not a model run touchstone, need to extract coverage set directly
+    cov_sets2 <- DBI::dbGetQuery(con, "SELECT coverage_set.id AS coverage_set, vaccine, activity_type, gavi_support_level
+                                FROM coverage_set
+                                WHERE touchstone = $1
+                                AND gavi_support_level != 'none'", touchstone_cov)
+    cov_sets2 <- merge_by_common_cols(disease_vaccine_delivery, cov_sets2, all.x = TRUE)
+    cov_sets2$scenario_type <- "default"
+    cov_sets <- cov_sets2[names(cov_sets)]
+  }
+  country_ <- ifelse(is.null(countries_to_extract), 
+                     "", 
+                     sprintf("AND country IN %s", sql_in(countries_to_extract, text_item = TRUE)))
+  cov <- DBI::dbGetQuery(con, sprintf("SELECT coverage_set, country, year, age_from, age_to, gender.name AS gender, gavi_support, target, coverage 
+                                      FROM coverage 
+                                      JOIN gender ON gender.id = gender
+                                      WHERE coverage_set IN %s
+                                      AND coverage > 0
+                                      AND year IN %s
+                                      %s",
+                                      sql_in(cov_sets$coverage_set, text_item = FALSE),
+                                      sql_in(year_min:year_max, text_item = FALSE),
+                                      country_))
+  
+  cov <- merge_by_common_cols(cov_sets, cov)
+  cov$coverage_set <- NULL
+  
+  print("Extracted raw coverage data...")
+  
+  ## transform coverage data
+  cov <- unique(cov) # this is needed as we used to create multiple coverage_sets for MCV1 and DTP3 for LiST model
+  cov$activity_id <- seq_along(cov$vaccine) # attach an id to avoid combining national target population from multiple sias
+  
+  i <- cov$age_from > cov$age_to
+  if (any(i)){
+    message("---> detected age_from > age_to")
+    cov$age_from_tmp <- cov$age_from
+    cov$age_from[i] <- cov$age_to[i]
+    cov$age_to[i] <- cov$age_from_tmp[i]
+    cov$age_from_tmp <- NULL
+    message("---> swaped age_from and age_to as a correction")
+  }
+  
+  ## spliting coverage data by age groups
+  cov1 <- cov[cov$activity_type == "routine", ]
+  cov1$age <- cov1$age_from
+  cov2 <- cov[cov$activity_type == "campaign", ]
+  
+  d <- list(NULL)
+  for(i in seq_along(cov2$activity_id)){
```

Comment:
This block is just to split every age range e.g. 10 - 14 into 5 separate rows one each for age 10, 11, 12, 13, 14?